src/dashboard/apps/consent/models.py

```diff
@@ -57,7 +59,24 @@ def __str__(self):  # noqa: D105
         return f"{self.delivery_point} - {self.updated_at}: {self.status}"
 
     def save(self, *args, **kwargs):
-        """Update the revoked_at timestamps if the consent is revoked."""
+        """Saves with custom logic applied before persistency.
+
+        - If the consent status is revoked, `revoked_at` is updated to the current time.
+        - When creating a new consent instance, if the `start` and `end` dates are not
+        provided, they are set to default values: the `start` date is set to the current
+        date, and the `end` date is set to a computed date based on
+        `CONSENT_NUMBER_DAYS_END_DATE`.
+        """
+        # Update the `revoked_at` timestamps if the consent is revoked.
         if self.status == REVOKED:
             self.revoked_at = timezone.now()
+
+        # if the current instance is newly created,
+        # set default values for `start` and end `dates` if they are not provided.
+        if self._state.adding:
+            self.start = self.start or timezone.now()
+            self.end = self.end or calculate_consent_end_date(
+                CONSENT_NUMBER_DAYS_END_DATE
+            )
```

Comment:
Ok, you can set the `consent_end_date` as a default callable for the `end` field.